### PR TITLE
Tank Transfer Valve Cleanup and In-Hands

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Tanks Transfer Valves now have an in-hand sprite. Sprites from TG.

img

Additionally, tank transfer valves with large tanks (this really only applies to jetpacks at the moment) can no longer fit in backpacks.

Code Wise
- Cleans up implied vars
- Standard `obj/item/I` as opposed to `obj/item/item` in the `attackby` proc
- `forceMove` as opposed to `loc` setting
- Proper new lines
- TTVs should now properly garbage collect
- Fixes being able to attach NODROP tanks to TTVs

:cl: Fox MCCloud
tweak: Tank transfer valves have in-hand icons
tweak: Tank transfer valves with large tanks (ie: jetpacks) cannot fit in backpacks
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/transfer_valve.dm [hunk 1]
```diff
@@ -113,36 +118,40 @@
...
 		tank_two = null
 		update_icon()
+		if((!tank_one || tank_one.w_class < 4) && (w_class > 3))
+			w_class = 3
 	else if(href_list["open"])
 		toggle_valve()
 	else if(attached_device)
 		if(href_list["rem_device"])
-			attached_device.loc = get_turf(src)
-			attached_device:holder = null
```
Comment (Fox-McCloud) [inline code/game/objects/items/devices/transfer_valve.dm]: Reeee; why wasn't this typecast for an assembly when `attached_device` can *only* be an assembly?

File: code/game/objects/items/devices/transfer_valve.dm [hunk 2]
```diff
@@ -206,17 +215,12 @@
...
 
-	else if(valve_open==1 && (tank_one && tank_two))
+	else if(valve_open && tank_one && tank_two)
 		split_gases()
 		valve_open = 0
-		src.update_icon()
-
-// this doesn't do anything but the timer etc. expects it to be here
-// eventually maybe have it update icon to show state (timer, prox etc.) like old bombs
-/obj/item/device/transfer_valve/proc/c_state()
```
Comment (Fox-McCloud) [inline code/game/objects/items/devices/transfer_valve.dm]: This must be ancient.

Comment (tigercat2000) [inline code/game/objects/items/devices/transfer_valve.dm]: >and you delete it

Comment (Fox-McCloud) [inline code/game/objects/items/devices/transfer_valve.dm]: The comment? No; this bit of code? Yes.

Comment (tigercat2000) [inline code/game/objects/items/devices/transfer_valve.dm]: well you deleted the comment above this code

or are you just talking about all the comments in general

Comment (Fox-McCloud) [inline code/game/objects/items/devices/transfer_valve.dm]: https://github.com/ParadiseSS13/Paradise/pull/6349/files#diff-dcb33c6ba2f9f375aec134af5ca806a1R195

This is the only known comment ExaDev ever left.

Comment (tigercat2000) [inline code/game/objects/items/devices/transfer_valve.dm]: Oh, I see. My bad 😛

**Comment by Bigtimetripod** [comment]
So it no longer also fits into smuggler bags? RIP stealth bombs...

**Comment by taukausanake** [comment]
Normal tank bombs can still fit in bags, yes? The types you make from Toxins. You did specify jetpacks only but it's good to make sure of these things.
Does this also imply a weight class for tank transfer bombs depending on what tanks are used? If I used emergency tanks would it be smaller?

**Comment by Fox-McCloud** [comment]
>Normal tank bombs can still fit in bags, yes? The types you make from Toxins. You did specify jetpacks only but it's good to make sure of these things.
Does this also imply a weight class for tank transfer bombs depending on what tanks are used? If I used emergency tanks would it be smaller?

If the tank's weight class is bigger than the TTV device, then the TTV device will take on the tank's weight class. If the tank is less or equal weight class to the TTV, then the TTV's weight class is unaltered.

It really doesn't change anything, at all, unless you were fond of making jetpack TTV bombs.

So yes, you can still fit them in your backpack 99% of the time.

**Comment by taukausanake** [comment]
I never knew you could make jetpack tank bombs but I see how it would be implied. I know what I'm doing in my next Toxins run

**Comment by KasparoVy** [comment]
You maxcap with regular tanks, I don't see why you'd use jetpacks outside of trying to see how insanely high you can make the theoretical yield

**Comment by taukausanake** [comment]
@KasparoVy That is literally my mindset in everything. If I can make the biggest hole or completely wipe out the test range then I know I have hit maxcap

**Pull request merged**